### ISSUE TITLE
[new feature] Sharing settings on GitLab

### ISSUE THREAD

**Issue by alicerocheman** [shanalikhan/code-settings-sync#554]
My new team has all its projects on gitlab.

Is there any way your tool could be improved with gitlab compatibility?

Thanks for the good work :)

**Comment by shanalikhan** [comment]
Duplicate of #443

**Comment by shanalikhan** [comment]
Actually i dont have time nowadays to work on extension.
I'm looking for contributors to work with me in integration GitLab in Settings Sync!

Feel free to ask your teams if someone is willing to contribute in open source extension 
I will be more happy to assist. 

Thanks!

**Comment by paradygm** [comment]
> eel free to ask your teams if someone is willing to contribute in open source extension
> I will be more happy to assist.

I'm considering contributing to implement this feature which I think might interest lots of devs. I just need to allow some time for this. Will keep you informed.

**Comment by shanalikhan** [comment]
> I'm considering contributing to implement this feature which I think might interest lots of devs. I just need to allow some time for this. Will keep you informed.

@paradygm 

Have a look on `v3.4.4` code base - that aims to refactor the exsiting Github Gist backend with factory patterns to allow multiple backend in the future. you can checkout that on going branch and test and improve the existing code as well. Once the refactoring is completed, in order to add new backend - we simple need to add new classes and use them via adaptor factory.

There is a paid issue for about 200$+ for adding GitHub or GitLab backend as well once refactoring is complete, you can start work on that one too while refactoring the existing code branch
